Move dotfiles from APP_BASE

hfaulds opened the pull request:
At the moment files such as `${APP_BASE}/.file` aren't moved into the root of the project. This can cause build failures if those files are required.

hfaulds committed:
move dotfiles

hfaulds commented on bin/compile:
```diff
@@ -23,7 +23,7 @@ APP_DEPS=$(cat ${ENV_DIR}/APP_DEPS)       # ex. `api core`
 (
     stage=`mktemp -d`                                                          && # make a temp directory to stage "correct" layout
-    mv ${BUILD_DIR}/${APP_BASE}/* ${stage}                                     && # stage the app folder we care about
+    mv ${BUILD_DIR}/${APP_BASE}/{,.[!.]}* ${stage}                             && # stage the app folder we care about
```
This is quite terse. The brackets expand to make this:

```
mv ${BUILD_DIR}/${APP_BASE}/* ${BUILD_DIR}/${APP_BASE}/.[!.]* ${stage}
```

And `/.[!.]*` is `/.*` but not `/..*`

jdleesmiller reviewed (commented):
I notice that a few lines later, we set `shopt -s dotglob`. Would it make sense to just move that up to the top, so the `*` here would also match the dot files?

hfaulds commented:
@jdleesmiller good point, I split apart that line and moved it so we still find error if any of the files in `APP_DEPS` aren't actually there. Did a test deploy and it works great.

jdleesmiller reviewed (approved):
🚢
